### ISSUE TITLE
Please support new model..."LLaDA-Image"

### ISSUE THREAD

**Issue by thaivb** [comfyanonymous/ComfyUI#16088]
### Feature Idea

https://github.com/inclusionAI/LLaDA-Image

**Comment by Milor123** [comment]
+1

**Comment by T8mars** [comment]
I am preparing a native ComfyUI Core implementation and would like maintainer feedback before opening the large code PR.

### Pinned references

- Upstream implementation: inclusionAI/LLaDA-Image at `b4dfa9a3e50d90d6718975ca1fa1b0edbc90d512`
- Base weights: inclusionAI/LLaDA-Image at `e4e2703f410f7ddb6ee8d6b09dac6a8ec5093039`
- Turbo weights: inclusionAI/LLaDA-Image-Turbo at `f4afc52d925bbac4e22a1c947111fc1f127e37e5`
- License: Apache-2.0
- Each BF16 AIO checkpoint is approximately 49 GB.

### Proposed checkpoint contract

One AIO `.safetensors` file per variant (Base and Turbo remain separate):

```text
model.diffusion_model.*
text_encoders.llada2.*
text_encoders.queryformer.*
text_encoders.text_projection.*
text_encoders.sigvq.*
text_encoders.tokenizer_json
vae.*
```

`CheckpointLoaderSimple` returns separately managed `MODEL`, `CLIP`, and `VAE` objects. The safetensors `config` metadata contains every official component config plus variant/recommended sampling settings. A streaming converter validates the official shard indexes, component key spaces, tensor sizes/dtypes, fixed source revision, and a 29-file SHA-256 source lock for each variant; it then rereads and hashes the output.

### Native runtime/component ownership

- Diffusion transformer: native PyTorch, `comfy.ops`, Core optimized attention and Flux RoPE, `ModelType.FLOW`, `latent_formats.Flux2`.
- Conditioning patcher (`CLIP`): native LLaDA2 MoE backbone, QueryFormer, six-layer text projection, SigVQ, and embedded tokenizer. Expert banks use `comfy.ops.MoEExperts`; no Transformers/Diffusers/trust-remote-code/runtime network dependency.
- VAE: reuse Core's existing Flux2 VAE detection, 2x2 latent packing, batch-normalization, and managed encode/decode path.
- The model-specific conditioning extras are `attention_mask`, `semantic_features`, `semantic_mask`, and (editing only) `source_latents`.

### Minimal node graph

- Plain text mode: standard `CLIPTextEncode` positive/negative + `EmptyFlux2LatentImage`.
- VQ mode: `LLaDAImageVQConditioning` returns positive/negative conditioning; target latent remains standard `EmptyFlux2LatentImage`.
- Edit mode: `LLaDAImageEditConditioning` encodes one source image with SigVQ and VAE, applies semantics to positive only, attaches source latents to both CFG branches, and returns the matching target latent.
- `LLaDAImageScheduler` reads Base/Turbo from checkpoint metadata. Steps `0` selects the official default (Base 50, Turbo 4).
- Reuse `CFGGuider`, `RandomNoise`, `SamplerCustomAdvanced`, `VAEDecode`, and `SaveImage`.

### Sampling exception

Base uses the upstream Kumaraswamy-like pre-shift schedule, shift 1.0, and standard Euler (CFG 5.0). Turbo uses the upstream uniform pre-shift grid, rational shift 3.0, and a dedicated stochastic flow sampler (CFG 1.0):

```text
x0 = sample - sigma * model_output
next = (1 - sigma_next) * x0 + sigma_next * noise
```

Inside Comfy's sampler API the model callable already returns `x0`, so the native sampler performs the second line. Per-step noise uses Core's seeded noise source, avoiding upstream's implicit global-RNG behavior.

### Evidence completed so far

- Tiny reference-model parity against the pinned upstream implementation for text-to-image and editing transformer paths (`atol=rtol=2e-5`).
- QueryFormer, text projection, SigVQ pixel/codebook paths, LLaDA2 dense/MoE/GQA/partial-RoPE backbone, schedules, seeded Turbo sampling, CFG ownership, config/detection, and real temporary `.safetensors` loading through `CheckpointLoaderSimple` are covered by focused tests.
- LLaDA2 reference parity max absolute error was `2.384185791015625e-07` on the tiny configuration.
- Full BF16 GPU image parity and low-VRAM evidence are still completion gates, not claimed yet.

Could maintainers please confirm these three design choices before the PR is finalized?

1. One AIO file per variant with standard `MODEL` / `CLIP` / `VAE` separation.
2. QueryFormer, text projection, and SigVQ owned by the CLIP patcher because all are conditioning-time components.
3. A dedicated Turbo sampler, since standard Euler cannot reproduce the scheduler's per-step stochastic replacement.

**Comment by Milor123** [comment]
@T8mars With this, will you also add support for Int8 and int4 conversion? Or only the base fp16 and fp8?
